allenact/algorithms/onpolicy_sync/runner.py

```diff
@@ -264,31 +263,50 @@ def start_train(
         if num_workers > 1:
             distributed_port = find_free_port()
 
+        model_hash = None
         for trainer_it in range(num_workers):
+            training_kwargs = dict(
+                id=trainer_it,
+                checkpoint=checkpoint,
+                restart_pipeline=restart_pipeline,
+                experiment_name=self.experiment_name,
+                config=self.config,
+                results_queue=self.queues["results"],
+                checkpoints_queue=self.queues["checkpoints"]
+                if self.running_validation
+                else None,
+                checkpoints_dir=self.checkpoint_dir(),
+                seed=self.seed,
+                deterministic_cudnn=self.deterministic_cudnn,
+                mp_ctx=self.mp_ctx,
+                num_workers=num_workers,
+                device=devices[trainer_it],
+                distributed_port=distributed_port,
+                max_sampler_processes_per_worker=max_sampler_processes_per_worker,
+                initial_model_state_dict=initial_model_state_dict
+                if model_hash is None
+                else model_hash,
+            )
             train: BaseProcess = self.mp_ctx.Process(
-                target=self.train_loop,
-                kwargs=dict(
-                    id=trainer_it,
-                    checkpoint=checkpoint,
-                    restart_pipeline=restart_pipeline,
-                    experiment_name=self.experiment_name,
-                    config=self.config,
-                    results_queue=self.queues["results"],
-                    checkpoints_queue=self.queues["checkpoints"]
-                    if self.running_validation
-                    else None,
-                    checkpoints_dir=self.checkpoint_dir(),
-                    seed=self.seed,
-                    deterministic_cudnn=self.deterministic_cudnn,
-                    mp_ctx=self.mp_ctx,
-                    num_workers=num_workers,
-                    device=devices[trainer_it],
-                    distributed_port=distributed_port,
-                    max_sampler_processes_per_worker=max_sampler_processes_per_worker,
-                    initial_model_state_dict=initial_model_state_dict,
-                ),
+                target=self.train_loop, kwargs=training_kwargs,
             )
-            train.start()
+            try:
+                train.start()
+            except ValueError as e:
+                # If the `initial_model_state_dict` is too large we sometimes
```

Comment:
I assume in this state we don't need to join the process (not sure, but seems reasonable).